src/menu.rs

```diff
@@ -0,0 +1,195 @@
+#[derive(Debug, Clone)]
+pub struct Menu {
+  pub title: String,
+  pub items: Vec<MenuItem>,
+}
+
+impl Menu {
+  pub fn new(title: &str, items: Vec<MenuItem>) -> Self {
+    Self {
+      title: String::from(title),
+      items,
+    }
+  }
+}
+
+#[derive(Debug, Clone)]
+/// CustomMenu is a custom menu who emit an event inside the EventLoop.
+pub struct CustomMenu {
+  // todo: replace id by a type that
+  // we can send from the with_menu::<T>()
+  // or we could check to use the UserEvent::<T> but
+  // my latest test failed
+  pub id: String,
```

Comment:
Maybe `id` can be the hash of the `name` and users could retrieve a `MenuId` from the method.
This could also remove the usage of thread local on Windows.

Reply:
Sounds good! I wonder if they have the same `name` it can cause issue? 

Because, i wanted to use the same menu builder, and same event loop event for the tray app menu.

It may be harder to make different actions, if you have a menu  example `Quit` in the menu bar and `Quit` in the tray menu

Reply:
Hmm maybe we could give them different type like `MenuId` & `TrayId`.

Reply:
Done. Let me know what you think